src/netlink_pm_upstream.c

```diff
@@ -42,60 +43,426 @@
+                l_genl_msg_append_attr(
+                        msg,
+                        MPTCP_PM_ATTR_TOKEN,
+                        sizeof(token),  // sizeof(uint32_t)
+                        &token)
+                && append_local_addr_attr(msg, &local)
+                && append_remote_addr_attr(msg, &remote);
+
+        if (!appended) {
+                l_genl_msg_unref(msg);
+
+                return ENOMEM;
+        }
+
+        return l_genl_family_send(pm->family,
+                                  msg,
+                                  mptcpd_family_send_callback,
+                                  "add_subflow", /* user data */
+                                  NULL  /* destroy */) == 0;
 }
 
 static int upstream_remove_subflow(struct mptcpd_pm *pm,
                                    mptcpd_token_t token,
                                    struct sockaddr const *local_addr,
                                    struct sockaddr const *remote_addr)
 {
-        (void) pm;
-        (void) token;
-        (void) local_addr;
-        (void) remote_addr;
+        /*
+          Payload (nested):
+              Token
+              Local address family
+              Local address
+              Local port
+              Remote address family
+              Remote address
+              Remote port
 
-        return ENOTSUP;
+         */
+
+        struct addr_info local = {
+                .addr = local_addr,
+        };
+
+        struct addr_info remote = {
+                .addr = remote_addr,
+        };
+
+        size_t const payload_size =
+                MPTCPD_NLA_ALIGN(token)
+                + MPTCPD_NLA_ALIGN(sizeof(uint16_t))  // local family
+                + MPTCPD_NLA_ALIGN_ADDR(local_addr)
+                + MPTCPD_NLA_ALIGN(sizeof(uint16_t))  // local port
+                + MPTCPD_NLA_ALIGN(sizeof(uint16_t))  // remote family
+                + MPTCPD_NLA_ALIGN_ADDR(remote_addr)
+                + MPTCPD_NLA_ALIGN(sizeof(uint16_t));  // remote port
+
+        struct l_genl_msg *const msg =
+                l_genl_msg_new_sized(MPTCP_PM_CMD_SUBFLOW_DESTROY,
+                                     payload_size);
+
+        bool const appended =
+                l_genl_msg_append_attr(
+                        msg,
+                        MPTCP_PM_ATTR_TOKEN,
+                        sizeof(token),  // sizeof(uint32_t)
+                        &token)
+                && append_local_addr_attr(msg, &local)
+                && append_remote_addr_attr(msg, &remote);
+
+        if (!appended) {
+                l_genl_msg_unref(msg);
+
+                return ENOMEM;
+        }
+
+        return l_genl_family_send(pm->family,
+                                  msg,
+                                  mptcpd_family_send_callback,
+                                  "remove_subflow", /* user data */
+                                  NULL  /* destroy */) == 0;
 }
 
 static int upstream_set_backup(struct mptcpd_pm *pm,
```

Comment:
The `upstream_set_backup()` implementation is missing.  Rather than delay this pull request further, I'd like to create another pull request for the  `upstream_set_backup()` implementation.